src/core/TrainController.hpp

```diff
@@ -37,15 +36,15 @@ namespace godot {
              * because the mover initialization and state sharing routines can be changed in the future. */
 
             // TrainController mozna bedzie rozszerzac klasami pochodnymi i przeslaniac metody
-            void _do_update_internal_mover(TMoverParameters *mover);
-            void _do_fetch_state_from_mover(TMoverParameters *mover, Dictionary &state);
+            void _do_update_internal_mover(TMoverParameters *mover) const;
+            static void _do_fetch_state_from_mover(TMoverParameters *mover, Dictionary &state);
```

Comment:
to nie jest static

Reply:
Niby mozemy zrobic static bo nie ma odwolania do dynamicznej zawartosci, ale zaufam i usune statica wylaczjac inspeckej na tej linii

Reply:
ciekawe, bo clang ani gcc sie nie czepial. u mnie bylo/jest czysto nawet z warningami.

Reply:
ad rem - Architektonicznie to nie moze byc static, bo impl może mieć odwolania do obiektu (this). Zostawiamy to customowej implementacji klas pochodnych, których konstrukcji teraz nie znamy. Analogicznie jest z _do_update_internal_mover